pullman/storageproviders/pvc/provider.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pvcprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kserve/modelmesh-runtime-adapter/internal/util"
+	"github.com/kserve/modelmesh-runtime-adapter/pullman"
+)
+
+const (
+	// config fields
+	configPVCName      = "name"
+	configPVCMountBase = "pvc_mount_base"
+
+	// defaults
+	defaultPVCMountBase = "/pvc_mounts"
+)
+
+type pvcProvider struct {
+}
+
+// pvcProvider implements StorageProvider
+var _ pullman.StorageProvider = (*pvcProvider)(nil)
+
+func (p pvcProvider) GetKey(config pullman.Config) string {
+	// hash the values of all configurations that go into creating the client
+	// no need to validate the config here
+	pvcName, _ := pullman.GetString(config, configPVCName)
+	pvcMountBase, _ := pullman.GetString(config, configPVCMountBase)
+
+	return pullman.HashStrings(pvcName, pvcMountBase)
```

Comment:
Here you can just return `""` (empty string) ... since the same instance of the repository can be used for all PVCs, there is no need to distinguish them here.